tests/unit/test_energy_spectrum.py

```diff
@@ -271,5 +255,181 @@ def test_when_no_foil_is_true(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+
+            sum_spectrum, es, _ = self.create_sum_and_energy_spectrum(tmp_dir)
+
+            # Check if a created sum spectrum is correct
+            self.assertEqual(sum_spectrum.spectra, es)
+            self.assertEqual(sum_spectrum.sum_spectrum,
+                             self.created_sum_spectrum)
+        return
+
+    def test_sum_spectrum_removal(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sum_spectrum, es, _ = self.create_sum_and_energy_spectrum(tmp_dir)
+
+            removed_spectrum = {
+                "1H.ERD.0": es["1H.ERD.0"]
+            }
+
+            # Check if a deleted spectrum is really removed
+            sum_spectrum.delete_spectra(removed_spectrum)
+            self.assertEqual(sum_spectrum.sum_spectrum,
+                             self.sum_spectrum_after_removed_spectra)
+        return
+
+    def test_sum_spectra_addition(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sum_spectrum, es, _ = self.create_sum_and_energy_spectrum(tmp_dir)
+
+            sum_spectrum.add_or_update_spectra(self.new_spectra)
+
+            # Check if an added spectra is really added
+            self.assertEqual(sum_spectrum.sum_spectrum,
+                             self.sum_spectrum_with_added_spectra)
+        return
+
+    def test_empty_sum_spectrum(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sum_spectrum, es, mesu = self.create_sum_and_energy_spectrum(
+                tmp_dir)
+
+            empty_spectrum = SumEnergySpectrum(None,
+                                               mesu.get_energy_spectra_dir(),
+                                               SumSpectrumType.MEASURED)
+
+            # Check if an empty spectrum really is empty
+            self.assertEqual(empty_spectrum.sum_spectrum, None)
+        return
+
+    def test_spectrum_file_contents(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            sum_spectrum, es, _ = self.create_sum_and_energy_spectrum(
+                tmp_dir)
+
+            expected = self.expected_sum_spectrum_file_content
+
+            with sum_spectrum.sum_spectrum_path.open() as sum_spectrum_file:
+                sum_lines = sum_spectrum_file.read().splitlines()
+
+            self.assertEqual(sum_lines, expected)
+        return
+
+
+def run_spectra_calculation(mesu, tmp_dir: Path, **kwargs):
+    """Helper for running spectra calculations"""
+    det = mesu.get_detector_or_default()
+    det.update_directories(tmp_dir / "Detector")
+    resource_dir = utils.get_resource_dir()
+    cuts = [
+        resource_dir / "cuts.1H.ERD.0.cut",
+        resource_dir / "cuts.1H.ERD.1.cut",
+        resource_dir / "cuts.35Cl.RBS_Mn.0.cut",
+        resource_dir / "cuts.7Li.0.0.0.cut",
+    ]
+    es = EnergySpectrum.calculate_measured_spectra(
+        mesu, cuts, **kwargs, verbose=False
+    )
+    return es
```

Comment:
This helper isn't used in the SumEnergySpectrum tests anymore, so it could/should be moved back where it was.

Reply:
Oops, I messed up the comment range. This comment applies to the `run_spectra_calculation` helper.